File: backend/core/src/main/java/com/careconnect/model/PackageItem.java
package com.careconnect.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import java.time.OffsetDateTime;

@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class PackageItem {
    private String trackingNumber;
    private String sender;

    @JsonProperty("expectedDateIso")
    private OffsetDateTime expectedDeliveryDate;  // null if unknown

    @JsonProperty("actions")
    private ActionLinks actionLinks;
}

File: frontend/lib/features/usps/domain/models/package_item.dart
import 'package:care_connect_app/features/usps/domain/models/action_links.dart';

class PackageItem {
  final String trackingNumber;
  final String? sender;
  final String? expectedDateIso;
  final ActionLinks actions;

  const PackageItem({
    required this.trackingNumber,
    this.sender,
    this.expectedDateIso,
    required this.actions,
  });
}
